_sources/Unit5-Writing-Classes/topic-5-1-parts-of-class.rst

```diff
@@ -37,49 +37,131 @@
 Anatomy of a Java Class
 =======================
 
-In Unit 2, we learned to use **classes** and **objects** that are built-in to Java or written by other programmers. In this unit, you will learn to write your own classes and objects!
+In Unit 2, we learned to use **classes** and **objects** that are built-in to
+Java or written by other programmers. In this unit, you will learn to write your
+own classes and make your own objects!
 
-Creating a Class
-------------------
+Remember that a **class** in Java defines a blueprint for creating objects. When
+you create **objects**, you create new **instances** of that class and what you
+can do with those instances is determined by what methods are defined in the
+class. Also recall that objects are always accessed by **references**—the values
+we actually store in variables or pass when we call methods are references to
+the objects so we can have different variables that refer to the same object.
 
-Remember that a **class** in programming defines a new **abstract data type**. When you create **objects**, you create new variables or **instances** of that class data type. For example in Unit 2, we created yertle and myrtle, 2 turtle objects created from the class Turtle, and we used the Java String class to create different String variables, also called **object references**.
+For example in Unit 2, we created ``yertle`` and ``myrtle``, 2 ``Turtle``
+variables and assigned them references to objects created from the class
+``Turtle`` and we used instances of Java’s ``String`` class to assign values to
+different ``String`` variables.
 
-Watch this short video to review the vocabulary in object-oriented programming:
+Watch this short video to review the vocabulary of object-oriented programming:
 
 .. youtube:: LfSaSANJPLg
     :width: 650
     :height: 415
     :align: center
 
-You can think of a class as like a blueprint or a cookie cutter or a factory that produces objects. For example, the class House below is a blueprint of a house and the objects which are individual houses built from that blueprint. You can't live on a blueprint although it describes a house! But you can use it to build houses and live in those.
 
-To write your own class, you typically start a class declaration with ``public`` then ``class`` then the name of the class.  The body of the class is defined inside a ``{`` and a ``}``. For example, the class House below. Then, you can create objects of that new House type by using ``Classname objectname = new Classname();``
+Creating a Class
+------------------
+
+Most classes you write will start with the keyword ``public`` though it is not
+always required. But the class definition proper always starts with the word
```

Comment:
Delete "proper"? Should we mention that the classname should match the filename, although they won't see that in Runestone?